<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Lujun Zhang and Jun Chen" />

<meta name="date" content="2024-01-03" />

<title>DiSC: A statistical tool for differential expression analyis of individual level single-cell RNA-Seq data</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">DiSC: A statistical tool for differential
expression analyis of individual level single-cell RNA-Seq data</h1>
<h4 class="author">Lujun Zhang and Jun Chen</h4>
<h4 class="date">2024-01-03</h4>



<div id="setups-and-introduction" class="section level1">
<h1>1. Setups and Introduction</h1>
<p>This vigenette will showcase how to use DiSC.</p>
<div id="required-dataset" class="section level2">
<h2>1.1. Required Dataset</h2>
<ul>
<li>This dataset was generated in the “Generate Simulation Datasets”
step in the <a href="https://github.com/Lujun995/DiSC_Replication_Code">“Parametric_simulation.rmd”</a>.</li>
<li>Coming from n12_12_d1.3_c375_vFALSE_0811_b1.RData</li>
<li>It contains 12 cases and 12 controls, each with 375 cell
replicates.</li>
<li>The read depths of each cell replicate are well-balanced.</li>
<li>A covariate at the individual level called RIN (RNA Integrity
Number) is included in the dataset.</li>
<li>The dataset comprises a total of 8,000 genes.</li>
<li><strong>Signal density:</strong> 15%, with differences in mean,
variance, and mean+variance (each at 5%). The ground truth of
differential/equally expression genes are indicated by
<code>gene_index</code>, including:
<ul>
<li><code>mean_index</code>: genes with a difference in mean</li>
<li><code>var_index</code>: genes with a difference in variance</li>
<li><code>mean_var_index</code>: genes with a difference in both mean
and variance</li>
<li><code>EE_index</code>: otherwise (to estimate type-I error)</li>
</ul></li>
<li>Difference in mean and/or variance = 1.3 folder change</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(DiSC)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">require</span>(GUniFrac)</span></code></pre></div>
</div>
</div>
<div id="data-input" class="section level1">
<h1>2. Data Input</h1>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sim_data&quot;</span>, <span class="at">package =</span> <span class="st">&quot;DiSC&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>count_matrix <span class="ot">&lt;-</span> sim_data<span class="sc">$</span>count_matrix</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>meta_cell <span class="ot">&lt;-</span> sim_data<span class="sc">$</span>meta_cell</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>gene_index <span class="ot">&lt;-</span> sim_data<span class="sc">$</span>gene_index</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>meta_ind <span class="ot">&lt;-</span> sim_data<span class="sc">$</span>meta_ind</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="fu">dim</span>(count_matrix)</span></code></pre></div>
<pre><code>## [1] 8000 9000</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">dim</span>(meta_cell)</span></code></pre></div>
<pre><code>## [1] 9000    4</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">dim</span>(meta_ind)</span></code></pre></div>
<pre><code>## [1] 24  3</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>count_matrix[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span></code></pre></div>
<pre><code>##       cell1 cell2 cell3 cell4 cell5 cell6
## gene1     1     0     0     0     0     1
## gene2     0     4     2     2     5     1
## gene3     2     0     4     0     0     2</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">head</span>(meta_cell)</span></code></pre></div>
<pre><code>##       cell_id individual phenotype cell_rd
## cell1   cell1       ind1         1   10705
## cell2   cell2       ind1         1   11888
## cell3   cell3       ind1         1   10926
## cell4   cell4       ind1         1   10314
## cell5   cell5       ind1         1   10623
## cell6   cell6       ind1         1   10075</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">head</span>(meta_ind)</span></code></pre></div>
<pre><code>##      individual phenotype         RIN
## ind1       ind1         1 -0.84838058
## ind2       ind2         1 -0.01896186
## ind3       ind3         1  0.52163194
## ind4       ind4         1  0.09235020
## ind5       ind5         1  0.52302505
## ind6       ind6         1 -1.77730860</code></pre>
</div>
<div id="disc-a-statistical-tool-for-differential-expression-analyis-of-individual-level-single-cell-rna-seq-data" class="section level1">
<h1>3. DiSC: A statistical tool for differential expression analyis of
individual level single-cell RNA-seq data</h1>
<p>DiSC currently relies on rarefaction to adjust for different
sequencing depths in each cell. The sequencing depth of each cell will
be rarefied to the minimal value across all cells. In cases where the
sample size is large, some cells may have very shallow sequencing
depths. In such instances, it is advisable to discard the several
lowest-sequenced samples. Our supplementary analyses showed that DiSC is
generally robust to rarefaction when discarding the least 0%-10%
sequenced cells.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>read_depth <span class="ot">=</span> <span class="fu">rowSums</span>(count_matrix)</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>discarding_flag <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="cf">if</span>(discarding_flag){</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  <span class="co"># when the sequencing depths of some samples are shallow</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>  depth <span class="ot">=</span> <span class="fu">min</span>(<span class="dv">5000</span>, <span class="fu">round</span>(<span class="fu">quantile</span>(read_depth, <span class="at">probs =</span> <span class="fl">0.1</span>)))</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;read depth filtering with threshold&quot;</span>, depth, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>  count_matrix_temp <span class="ot">&lt;-</span> count_matrix[, <span class="fu">which</span>(read_depth <span class="sc">&gt;=</span> depth)]</span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a>  meta_cell_temp <span class="ot">&lt;-</span> meta_cell <span class="sc">%&gt;%</span> </span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a>    <span class="fu">filter</span>(cell_id <span class="sc">%in%</span> <span class="fu">colnames</span>(count_matrix_temp))</span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a>  rarefy_mat <span class="ot">&lt;-</span> <span class="fu">t</span>(GUniFrac<span class="sc">::</span><span class="fu">Rarefy</span>(<span class="fu">t</span>(count_matrix_temp))<span class="sc">$</span>otu.tab.rff)</span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a>  <span class="co"># when all samples are sufficiently sequenced</span></span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a>  rarefy_mat <span class="ot">&lt;-</span> <span class="fu">t</span>(GUniFrac<span class="sc">::</span><span class="fu">Rarefy</span>(<span class="fu">t</span>(count_matrix))<span class="sc">$</span>otu.tab.rff)</span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a>}</span></code></pre></div>
<p>List of accepted parameters in function <code>DiSC</code>:</p>
<ul>
<li><code>ct.mat</code>: sequencing <strong>count</strong> matrix. Genes
are in rows, <strong>column names are cell ids</strong>.</li>
<li><code>cell.ind</code>: a data frame includes two columns of cell ids
and individual ids. It links cell ids to individual ids.</li>
<li><code>metadata</code>: a data frame includes a column of individual
id, a column of an outcome of interest and columns of covariates if
applicable.</li>
<li><code>outcome</code>: a character string which is the variable name
of outcome in <code>metadata</code>.</li>
<li><code>covariates</code>: a character string or vector of character
strings which are the variable names of covariates in
<code>metadata</code>. Default: <code>NULL</code>.</li>
<li><code>stats.combine.func</code>: a function, method to combine
F-statistics using different features, default <code>max</code>.</li>
<li><code>cell.id</code>: a character string which is the variable name
of cell ids in <code>cell.ind</code>.</li>
<li><code>individual.id</code>: a character string which is the variable
name of the individual id variables in <code>cell.ind</code> and
<code>metadata</code>.</li>
<li><code>perm.no</code>: an integer, number of permutations used.
Default: 999. Can be reduced to 99 if adjusted P-values (false discovery
rate) is of the only interest.</li>
<li><code>features</code>: Vector of a character string. Choose from
“prev” (zero/non-zero proportions), “nzm” (non-zero mean), “nzsd”
(non-zero standard deviation), “mean” (overall mean), “sd” (overall
standard deviation). Features used to test for the differentially
expressed genes. Default: “prev”, “nzm” and “nzsd”.</li>
<li><code>verbose</code>: Logical. Should the function print the
processes? Default: <code>TRUE</code>.</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>obj1 <span class="ot">&lt;-</span> <span class="fu">DiSC</span>(<span class="at">ct.mat =</span> rarefy_mat, <span class="at">cell.ind =</span> meta_cell,</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>             <span class="at">metadata =</span> meta_ind, <span class="at">outcome =</span> <span class="st">&quot;phenotype&quot;</span>,</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>             <span class="at">covariates =</span> <span class="st">&quot;RIN&quot;</span>, <span class="at">cell.id =</span> <span class="st">&quot;cell_id&quot;</span>,</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>             <span class="at">individual.id =</span> <span class="st">&quot;individual&quot;</span>, <span class="at">perm.no =</span> <span class="dv">999</span>,</span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>             <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&#39;prev&#39;</span>, <span class="st">&#39;nzm&#39;</span>, <span class="st">&#39;nzsd&#39;</span>), <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## The data has  24  samples and  8000  genes will be tested!
## Permutation testing ...
## ...................................................................................................
## *Completed!</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># Type I error</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="fu">mean</span>(obj1<span class="sc">$</span>p.raw[gene_index<span class="sc">$</span>EE_index] <span class="sc">&lt;=</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## [1] 0.03455882</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="co"># Power</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="fu">mean</span>(obj1<span class="sc">$</span>p.raw[gene_index<span class="sc">$</span>mean_index] <span class="sc">&lt;=</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## [1] 0.5375</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="fu">mean</span>(obj1<span class="sc">$</span>p.raw[gene_index<span class="sc">$</span>var_index] <span class="sc">&lt;=</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## [1] 0.7625</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="fu">mean</span>(obj1<span class="sc">$</span>p.raw[gene_index<span class="sc">$</span>mean_var_index] <span class="sc">&lt;=</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## [1] 0.9025</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="co"># False discovery proportion</span></span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="fu">sum</span>(obj1<span class="sc">$</span>p.adj.fdr[gene_index<span class="sc">$</span>EE_index] <span class="sc">&lt;=</span> <span class="fl">0.10</span>)<span class="sc">/</span></span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>  <span class="fu">sum</span>(obj1<span class="sc">$</span>p.adj.fdr <span class="sc">&lt;=</span> <span class="fl">0.10</span>)</span></code></pre></div>
<pre><code>## [1] 0.05714286</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="co"># Number of positive discoveries</span></span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a><span class="fu">sum</span>(obj1<span class="sc">$</span>p.adj.fdr[gene_index<span class="sc">$</span>mean_index] <span class="sc">&lt;=</span> <span class="fl">0.10</span>)</span></code></pre></div>
<pre><code>## [1] 125</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="fu">sum</span>(obj1<span class="sc">$</span>p.adj.fdr[gene_index<span class="sc">$</span>var_index] <span class="sc">&lt;=</span> <span class="fl">0.10</span>)</span></code></pre></div>
<pre><code>## [1] 212</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="fu">sum</span>(obj1<span class="sc">$</span>p.adj.fdr[gene_index<span class="sc">$</span>mean_var_index] <span class="sc">&lt;=</span> <span class="fl">0.10</span>)</span></code></pre></div>
<pre><code>## [1] 323</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>t <span class="ot">&lt;-</span> <span class="fu">proc.time</span>()</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>obj2 <span class="ot">&lt;-</span> <span class="fu">DiSC</span>(<span class="at">ct.mat =</span> rarefy_mat, <span class="at">cell.ind =</span> meta_cell,</span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>             <span class="at">metadata =</span> meta_ind, <span class="at">outcome =</span> <span class="st">&quot;phenotype&quot;</span>,</span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>             <span class="at">covariates =</span> <span class="st">&quot;RIN&quot;</span>, <span class="at">cell.id =</span> <span class="st">&quot;cell_id&quot;</span>,</span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a>             <span class="at">individual.id =</span> <span class="st">&quot;individual&quot;</span>, <span class="at">perm.no =</span> <span class="dv">99</span>,</span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a>             <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&#39;prev&#39;</span>, <span class="st">&#39;nzm&#39;</span>, <span class="st">&#39;nzsd&#39;</span>), <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## *</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a><span class="fu">print</span>(t <span class="sc">-</span> <span class="fu">proc.time</span>())</span></code></pre></div>
<pre><code>##    user  system elapsed 
##   -9.52   -1.73  -19.93</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>obj1<span class="sc">$</span>p.adj.fdr[<span class="dv">123</span>]</span></code></pre></div>
<pre><code>##   gene123 
## 0.8862973</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a>obj2<span class="sc">$</span>p.adj.fdr[<span class="dv">123</span>]</span></code></pre></div>
<pre><code>##   gene123 
## 0.8801008</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="fu">sum</span>(obj1<span class="sc">$</span>p.adj.fdr[gene_index<span class="sc">$</span>mean_index] <span class="sc">&lt;=</span> <span class="fl">0.10</span>)</span></code></pre></div>
<pre><code>## [1] 125</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a><span class="fu">sum</span>(obj1<span class="sc">$</span>p.adj.fdr[gene_index<span class="sc">$</span>var_index] <span class="sc">&lt;=</span> <span class="fl">0.10</span>)</span></code></pre></div>
<pre><code>## [1] 212</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="fu">sum</span>(obj1<span class="sc">$</span>p.adj.fdr[gene_index<span class="sc">$</span>mean_var_index] <span class="sc">&lt;=</span> <span class="fl">0.10</span>)</span></code></pre></div>
<pre><code>## [1] 323</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
